src/Microsoft.SqlTools.ServiceLayer/Scripting/ScripterCore.cs

```diff
@@ -794,16 +795,23 @@ internal string SelectFromTableOrView(Server server, Urn urn, bool isDw)
             // In Hekaton M5, if it's a memory optimized table, we need to provide SNAPSHOT hint for SELECT.
             if (urn.Type.Equals("Table") && ScriptingUtils.IsXTPSupportedOnServer(server))
             {
-                Table table = (Table)server.GetSmoObject(urn);
-                table.Refresh();
-                if (table.IsMemoryOptimized)
+                try
                 {
-                    selectQuery.Append(" WITH (SNAPSHOT)");
+                    Table table = (Table)server.GetSmoObject(urn);
```

Comment:
this is throwing exceptions for both Azure DB and servers with non-standard ports.  Parsing out the port avoids the exception for on-prem servers.  I'm not sure what to about Azure DB.  It looks like the URN parser what's the server to not include ".database.window.net", but I didn't investigate.  Swallowing the exception is probably fine in most cases, as InMemory tables in Azure DB are currently rare and the user impact is the SELECT won't include WITH (SNAPSHOT) in those rare cases.